### ISSUE TITLE
Use Sphinx ref links or URLs in docstrings?

### ISSUE THREAD

**Issue by cdeil** [astropy/astropy#9580]
I noticed that there's some raw URL links pointing to http://docs.astropy.org/en/stable from class docstrings such as `astropy.table.Table`:

https://github.com/astropy/astropy/blob/e22110689b7bd95d7e674b720620e07b97d2d714/astropy/table/table.py#L338

or `astropy.uncertainty.Distribution`:

https://github.com/astropy/astropy/blob/297589cd9b3f087795803cd698d8410c46cac0eb/astropy/uncertainty/core.py#L30

This can be problematic, since it takes users to a different docs version, when reading an older docs version. Since Astropy isn't changing much, in practice it's probably OK, but it could happen that links break, and generally I think users will be surprised if docs version changes when clicking around in the docs.

Most other projects use Sphinx ref links to generate relative links in the HTML (or PDF if generated) within a given docs version. Example:

https://github.com/scikit-learn/scikit-learn/blob/1495f69242646d239d89a5713982946b8ffcf9d9/sklearn/linear_model/ridge.py#L599

Should we change to relative links?
(I'm +1 to do that, and could send a PR if it's the way to go.)

Or was there a deliberate decision to put raw URLs to http://docs.astropy.org/en/stable ?
(there is the advantage that from the help docstring in `python`, `ipython`, `Jupyter` one can get the URL quickly)

**Comment by bmorris3** [comment]
Just chiming in to say that I find the raw URLs really handy – they're the easiest way to find the rendered docs corresponding to the docstring I'm looking at.

**Comment by pllim** [comment]
I vaguely remember bringing such a thing up in the past but @taldcroft or @mhvk (maybe) said they preferred an explicit link like that so people can go to the link from `help(...)` interface in a Python session because Sphinx ref link only renders in HTML doc.

**Comment by bsipocz** [comment]
Yes, indeed this was pointed out at the time these links ended up in the docs: https://github.com/astropy/astropy/pull/8249#pullrequestreview-182927386

I see the point why they ended up in there, while I would indeed prefer to have a self consistent docs.

**Comment by cdeil** [comment]
OK, so some devs prefer one and some the other, and currently searching for `docs.astropy.org` and `:ref:` in Astropy docstrings shows that it's a mix, reflecting the two preferences. Like I said, I would change to `:ref:` everywhere, but I don't think it important enough to argue about, so I'm closing this issue.
